clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java

```diff
@@ -178,7 +179,21 @@ private void initInternalStorageEngineFactories() {
     persistenceTypeToStorageEngineFactoryMap.put(BLACK_HOLE, new BlackHoleStorageEngineFactory());
   }
 
-  private void restoreAllStores(
+  private boolean deleteStorageEngine(String storageEngineName) {
```

Comment:
For unit test, I think you can turn this function into a `static` method, which will take two params:
```
readOnlyStoreRepository
storageEngineName
```
And you can write unit tests against it.

Also, I think we better to rename it to be something like: `whetherToDeleteStorageEngineWhenEncounterError` to make it less scary.